impl/src/main/java/org/ehcache/impl/serialization/ByteArraySerializer.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.impl.serialization;
+
+import org.ehcache.core.spi.service.FileBasedPersistenceContext;
+import org.ehcache.exceptions.SerializerException;
+import org.ehcache.spi.serialization.Serializer;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+/**
+ * Default {@link Serializer} for {@code byte[]} type. Simply writes the byte array
+ * to a byte buffer.
+ */
+public class ByteArraySerializer implements Serializer<byte[]> {
+
+  /**
+   * No arg constructor
+   */
+  public ByteArraySerializer() {
+  }
+
+  /**
+   * Constructor to enable this serializer as a transient one.
+   * <P>
+   *   Parameter is ignored as {@code byte[]} is a base java type.
+   * </P>
+   *
+   * @param classLoader the classloader to use
+   *
+   * @see Serializer
+   */
+  public ByteArraySerializer(ClassLoader classLoader) {
+  }
+
+  /**
+   * Constructor to enable this serializer as a persistent one.
+   * <P>
+   *   Parameters are ignored as {@code byte[]} is a base java type and this implementation requires no state.
+   * </P>
+   *
+   * @param classLoader the classloader to use
+   * @param persistenceContext the persistence context
+   *
+   * @see Serializer
+   */
+  public ByteArraySerializer(ClassLoader classLoader, FileBasedPersistenceContext persistenceContext) {
+
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ByteBuffer serialize(byte[] object) throws SerializerException {
+    return ByteBuffer.wrap(object);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public byte[] read(ByteBuffer binary) throws ClassNotFoundException, SerializerException {
+    byte[] bytes = new byte[binary.remaining()];
+    binary.get(bytes);
+    return bytes;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean equals(byte[] object, ByteBuffer binary) throws ClassNotFoundException, SerializerException {
+    return Arrays.equals(read(binary), object);
```

Comment:
This isn't the equals contract of arrays. (also wouldn't it be faster to do `binary.equals(serialize(object))`?)